test/exercises_test.dart

```diff
@@ -27,14 +27,37 @@ Future runCmd(List<String> cmds) async {
   assert(res.exitCode == 0);
 }
 
+Future getPackageName() async {
+  final pubspec = new File("pubspec.yaml");
+
+  final packageName = loadYaml(await pubspec.readAsString())["name"];
+
+  return packageName;
+}
+
+Future locateExercismDirAndExecuteTests() async {
+  final exercisesRootDir = new Directory("exercises");
+
+  assert(await exercisesRootDir.exists());
+
+  final exercisesDirs = exercisesRootDir.listSync().where((d) => d is Directory);
+
+  /// Sort directories alphabetically
+  final sortedExerciseDirs = exercisesDirs.toList();
+  sortedExerciseDirs.sort((a, b) => a.path.compareTo(b.path));
+
+  for (dynamic dir in sortedExerciseDirs) {
```

Comment:
Same point for using `dynamic`. See below comment
Use `FileSystemEntity`, since we are actually calling a getter `path` in the loop it's important that it actually exists.
The compiler won't check if it's a `dynamic` variable.